bot/exts/holidays/easter/easter_riddle.py

```diff
@@ -57,26 +60,33 @@ async def riddle(self, ctx: commands.Context) -> None:
 
         description = f"You have {TIMELIMIT} seconds before the first hint."
 
-        riddle_embed = discord.Embed(title=question, description=description, colour=Colours.pink)
-
-        await ctx.send(embed=riddle_embed)
-        await asyncio.sleep(TIMELIMIT)
-
-        hint_embed = discord.Embed(
-            title=f"Here's a hint: {hints[0]}!",
-            colour=Colours.pink
-        )
-
-        await ctx.send(embed=hint_embed)
-        await asyncio.sleep(TIMELIMIT)
-
-        hint_embed = discord.Embed(
-            title=f"Here's a hint: {hints[1]}!",
-            colour=Colours.pink
+        riddle_embed = discord.Embed(
+            title=question, description=description, colour=Colours.pink
         )
 
-        await ctx.send(embed=hint_embed)
-        await asyncio.sleep(TIMELIMIT)
+        await ctx.send(embed=riddle_embed)
+        hint_number = 0
+        while hint_number < 2:
+            try:
+                response = await self.bot.wait_for(
+                    "message",
+                    check=lambda m: m.channel == ctx.channel
+                    and m.author != self.bot.user,
+                    timeout=TIMELIMIT,
+                )
+                if response.content.lower() == self.correct.lower():
+                    self.winners.add(response.author.mention)
+                    break
+            except TimeoutError:
+                try:
+                    hint_embed = discord.Embed(
+                        title=f"Here's a hint: {hints[hint_number]}!",
+                        colour=Colours.pink,
+                    )
+                except IndexError:
+                    break
+                hint_number += 1
+                await ctx.send(embed=hint_embed)
```

Comment:
I don't think this will work as wanted, if you keep sending incorrect answers it will reset the timeout each time and keep running.